packages/web-console/src/lib/components/profiler/README.md

```diff
@@ -0,0 +1,194 @@
+│  ┌─────┐   │                        │
+│  │Mini │   │                        │
+│  │ map │   │                        │
+│  └─────┘   │                        │
+└────────────┴────────────────────────┘
+```
+
+**Sidebar** (left):
+
+- Metric selector dropdown (time%, memory, etc.)
+- Worker thread checkboxes
+- Navigator minimap
+
+**Main Area** (right):
+
+- Interactive graph with pan/zoom
+- Color-coded nodes (performance heatmap)
+- Hover tooltips with metrics and SQL source
+
+## Styling
+
+The component includes default styles but can be customized:
+
+```svelte
+<ProfilerDiagram {profileData} {dataflowData} class="custom-profiler" />
+
+<style>
+  :global(.custom-profiler) {
+    /* Override wrapper styles */
+  }
+
+  :global(.custom-profiler .profiler-graph) {
+    /* Override graph container */
+  }
+</style>
+```
+
+### CSS Variables (Future Enhancement)
+
+Consider adding CSS custom properties for theming:
+
+- `--profiler-bg-color`
+- `--profiler-border-color`
+- `--profiler-sidebar-width`
+
+## Data Requirements
+
+### Profile Data (`JsonProfiles`)
+
+Obtained from pipeline manager API:
+
+```typescript
+GET / v0 / pipelines / { pipeline_name } / circuit_profile
+```
+
+Contains:
+
+- `worker_profiles[]`: Per-worker performance measurements
+- `graph`: Circuit structure (nodes, edges, hierarchies)
+
+### Dataflow Graph (`Dataflow`)
+
+Generated by SQL compiler:
+
+```bash
+cd sql-to-dbsp-compiler
+./SQL-compiler/sql-to-dbsp \
+  --input program.sql \
+  --dataflow output.json
+```
+
+Contains:
+
+- `sources[]`: SQL source code lines
+- `mir`: MIR node operations with source positions
+- Mappings between SQL code and circuit operators
+
+**Note**: Currently requires manual compilation. Future enhancement: fetch from pipeline manager API.
+
+## Integration Example
+
+```svelte
+<script lang="ts">
+  import { onMount } from 'svelte'
+  import ProfilerDiagram from '$lib/components/profiler/ProfilerDiagram.svelte'
+  import type { JsonProfiles, Dataflow } from 'profiler-lib'
+
+  let profileData = $state<JsonProfiles | null>(null)
+  let dataflowData = $state<Dataflow | null>(null)
+  let loading = $state(true)
+  let error = $state<string | null>(null)
+
+  onMount(async () => {
+    try {
+      // Fetch both datasets in parallel
+      const [profile, dataflow] = await Promise.all([
+        fetch(`/v0/pipelines/${pipelineId}/circuit_profile`).then((r) => r.json()),
+        fetch(`/dataflow/${pipelineId}.json`).then((r) => r.json())
+      ])
+
+      profileData = profile
+      dataflowData = dataflow
+    } catch (e) {
+      error = e instanceof Error ? e.message : 'Failed to load profiler data'
+    } finally {
+      loading = false
+    }
+  })
+</script>
+
+{#if loading}
+  <div class="loading">Loading profiler...</div>
+{:else if error}
+  <div class="error">{error}</div>
+{:else if profileData && dataflowData}
+  <ProfilerDiagram {profileData} {dataflowData} />
+{/if}
+```
+
+## User Interactions
```

Comment:
this will need a separate documentation, which I will write; the repository is probably not the right place to host it